developer-workflow/extension-modules.rst

```diff
@@ -5,13 +5,571 @@
+
+For built-in extension modules, update :cpy-file:`Modules/Setup.bootstrap.in`
+by adding the following line after the ``*static*`` marker:
+
+.. code-block:: text
+   :caption: :cpy-file:`Modules/Setup.bootstrap.in`
+   :emphasize-lines: 3
+
+   *static*
+   ...
+   fastfoo _foo/_foomodule.c _foo/helper.c
+   ...
+
+For other extension modules, update :cpy-file:`Modules/Setup.stdlib.in`
+by adding the following line after the ``*@MODULE_BUILDTYPE@*`` marker
+but before the ``*shared*`` marker:
+
+.. code-block:: text
+   :caption: :cpy-file:`Modules/Setup.stdlib.in`
+   :emphasize-lines: 3
+
+   *@MODULE_BUILDTYPE@*
+   ...
+   @MODULE_FASTFOO_TRUE@fastfoo _foo/_foomodule.c _foo/helper.c
+   ...
+   *shared*
+
+The ``@MODULE_<NAME>_TRUE@<name>`` marker expects ``<NAME>`` to be the
+upper-cased module name ``<name>``. If the extension module must be built
+as a *shared* module, put the ``@MODULE_FASTFOO_TRUE@fastfoo`` line after
+the ``*shared*`` marker:
+
+.. code-block:: text
+   :caption: :cpy-file:`Modules/Setup.stdlib.in`
+   :emphasize-lines: 4
+
+   ...
+   *shared*
+   ...
+   @MODULE_FASTFOO_TRUE@fastfoo _foo/_foomodule.c _foo/helper.c
+
+Compiling the CPython project
+-----------------------------
+
+Now that everything is in place, it remains to compile the project:
+
+.. code-block:: shell
+
+   make regen-configure
+   ./configure
+   make regen-all
+   make regen-stdlib-module-names
+   make
+
+.. tip::
+
+   Use ``make -j12`` to speed-up compilation if you have enough CPU cores.
+
+* ``make regen-configure`` updates the :cpy-file:`configure` script.
+
+* ``make regen-all`` is responsible for regenerating header files and
+  invoking other scripts, such as :ref:`Argument Clinic <clinic>`.
+  Execute this rule if you do not know which files should be updated.
+
+* ``make regen-stdlib-module-names`` updates the standard module names, making
+  :mod:`!fastfoo` discoverable and importable via ``import fastfoo``.
+
+* The final ``make`` step is generally not needed since the previous ``make``
+  invokations may completely rebuild the project, but it could be needed in
+  some specific cases.
+
+Troubleshooting
+---------------
+
+This section addresses common issues that you may face when following this tutorial.
+
+No rule to make target ``regen-configure``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+This usually happens after running ``make distclean`` (which removes
+the ``Makefile``). The solution is to regenerate the :cpy-file:`configure`
+script as follows:
+
+.. code-block:: shell
+
+   ./configure            # for creating a Makefile
+   make regen-configure   # for updating 'configure'
+   ./configure            # for updating the Makefile
+
+``make regen-configure`` and missing permissions
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Since this rule requires `Docker <https://docs.docker.com/desktop>`_ to be
+running, the following can be done on Linux platforms (``systemctl``-based):
```

Comment:
Not sure if it's worth mentioning that instead of Docker you can use `podman`, which can be just installed, it doesn't need to be running.

Reply:
Well... I have no idea how to help people with podman :') but yes, I'll mention it as well. I'll have a look at it afterwards